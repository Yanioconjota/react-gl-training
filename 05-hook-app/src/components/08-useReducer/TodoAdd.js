import React from 'react';
import { Button } from "react-bootstrap";
import { useForm } from "../../hooks/useForm";

const TodoAdd = ({ handleAddTodo }) => {
  const [{ description }, handleInputChange, resetForm] = useForm({
    description: "",
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length < 1) {
      return;
    }
    const newTodo = {
      id: new Date().getTime(),
      desc: description,
      done: false,
    };

    handleAddTodo(newTodo);
    resetForm();
  };

  return (
    <>
      <h4>Agregar todo</h4>
      <hr />
      <form onSubmit={handleSubmit}>
        <input
          className="form-control"
          type="text"
          name="description"
          autocompplete="off"
          placeholder="Tengo que..."
          value={description}
          onChange={handleInputChange}
        />
        <div className="d-grid mt-2">
          <Button type="submit" variant="outline-primary">
            Agregar
          </Button>
        </div>
      </form>
    </>
  );
};

export default TodoAdd;
