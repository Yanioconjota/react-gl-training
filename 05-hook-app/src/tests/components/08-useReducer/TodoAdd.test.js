import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import TodoAdd from "../../../components/08-useReducer/TodoAdd";

const handleAddTodo = jest.fn();
describe('pruebas con TodoAdd', () => {
  
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);
  test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('no debe llamar al handleAddTodo si el input está vacío', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');
    console.log(formSubmit);
    formSubmit({preventDefault(){}});
    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test('debe llamar al handleAddTodo', () => {
    const value = 'Aprender Node';
    const input = wrapper.find('input');
    
    input.simulate('change', {
      target: {
        value: value,
        name: 'description'
      }
    });

    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({
      preventDefault(){}
    });
    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handleAddTodo).toHaveBeenCalledWith({
      desc: value,
      done: false,
      id: expect.any(Number)
    });
    expect(input.prop('value')).toBe('');
  });
  
  
});
