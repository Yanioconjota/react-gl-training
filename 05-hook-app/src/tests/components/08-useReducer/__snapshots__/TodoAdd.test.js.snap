// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con TodoAdd debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h4>
    Agregar todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autocompplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Tengo que..."
      type="text"
      value=""
    />
    <div
      className="d-grid mt-2"
    >
      <Button
        active={false}
        disabled={false}
        type="submit"
        variant="outline-primary"
      >
        Agregar
      </Button>
    </div>
  </form>
</Fragment>
`;
